src/passkey_child/passkey_child_assert.c

```diff
@@ -75,28 +75,59 @@ get_assert_user_id(TALLOC_CTX *mem_ctx, fido_assert_t *assert,
 }
 
 errno_t
-set_assert_client_data_hash(fido_assert_t *_assert)
+set_assert_client_data_hash(const struct passkey_data *data,
+                            fido_assert_t *_assert)
 {
+    TALLOC_CTX *tmp_ctx = NULL;
     unsigned char cdh[32];
+    unsigned char *crypto_challenge = NULL;
+    size_t crypto_challenge_len;
     errno_t ret;
 
-    ret = sss_generate_csprng_buffer(cdh, sizeof(cdh));
-    if (ret != EOK) {
-        DEBUG(SSSDBG_OP_FAILURE,
-              "sss_generate_csprng_buffer failed [%d]: %s.\n",
-              ret, fido_strerr(ret));
-        goto done;
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_new() failed.\n");
+        return ENOMEM;
     }
 
-    ret = fido_assert_set_clientdata_hash(_assert, cdh, sizeof(cdh));
-    if (ret != FIDO_OK) {
-        DEBUG(SSSDBG_OP_FAILURE,
-              "fido_assert_set_clientdata_hash failed [%d]: %s.\n",
-              ret, fido_strerr(ret));
-        goto done;
+    if (data->action == ACTION_AUTHENTICATE) {
+        ret = sss_generate_csprng_buffer(cdh, sizeof(cdh));
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                "sss_generate_csprng_buffer failed [%d]: %s.\n",
+                ret, fido_strerr(ret));
+            goto done;
+        }
+
+        ret = fido_assert_set_clientdata_hash(_assert, cdh, sizeof(cdh));
```

Comment:
_assert suggest it is an output variable but it is actually used as input

Reply:
No, this call sets the client data hash in the assert. And we are using the assert outside this function, making it an output variable.

Reply:
Ok, I misunderstood the man page.